### DISCUSSION TITLE
TV CRT monitor like the VGA component

### DISCUSSION THREAD

**Discussion by Digimorf** [hneemann/Digital#1233] [Ideas]
Good morning,

First of all, congratulations for the splendid product, which is at the same level as other professional digital circuit simulation tools.

I'm using digital as a reverse engineering tool for a personal project involving developing a replica of an Apple Machintosh 128k computer.

This computer has several PAL chips, which perform different tasks for the functioning of the entire system. To understand how these chips, and therefore the computer, behave, I am digitally recreating the different sections of the computer.

[https://macintosh-replica-story.blogspot.com](https://macintosh-replica-story.blogspot.com)

![Screenshot 2023-11-16 090532](https://github.com/hneemann/Digital/assets/22617118/08d1a664-0dc3-4c80-8da1-dee4f90c8a15)
![Screenshot 2023-11-16 091317](https://github.com/hneemann/Digital/assets/22617118/6f4c48a1-b139-493c-8f38-d5bb32b8780e)

I found the possibility of displaying graphic output on virtual VGA spectacular. Even though it's not real time it's a truly spectacular tool. Since the Macintosh was equipped with a monitor with a resolution of 512 by 342 and with custom timings, I am adapting the simulation to display the frame buffer inside the VGA frame thus causing the presence of a border.
                  
I was wondering if it was possible to adapt the VGA component to also handle other frequencies based on HSync and VSync.

Congratulations again and I will not stop supporting this tool.

**Comment by hneemann** [comment]
It should not be a problem to support another video mode. 
[There are already a few](https://github.com/hneemann/Digital/blob/master/src/main/java/de/neemann/digital/gui/components/graphics/VGA.java), so it's easy to add one. Where can I find the exact timing parameters?

**Reply by Digimorf** [reply]
Hi, Thank you for your kind reply.
it's awesome that this component supports all those video modes.
The video mode used by the Macintosh is well explained in this document that I have found on the Internet. The guy who drew this. schematic was able to interface a Macintosh Classic II monitor to his Beaglebone development board.

<img width="483" alt="timing" src="https://github.com/hneemann/Digital/assets/22617118/c94a8b08-6555-452d-b033-888e538a9c8c">

It could be interesting if it was possible to add those parameters directly into the VGA component settings, or somewhere else.
